spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/jdbc/DataSourceAutoConfigurationTests.java

```diff
@@ -158,6 +158,20 @@ void explicitTypeNoSupportedDataSource() {
 				.run(this::containsOnlySimpleDriverDataSource);
 	}
 
+	/**
+	 * This test makes sure that if no supported pool data source is present, a datasource
```

Comment:
If possible, when merging, we should replace this comment with a descriptive test method name